core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/DiagramServices.java

```diff
@@ -1479,6 +1484,39 @@ public void selectElementInDiagram(DSemanticDecorator newTarget) {
    * @param mapping
    * @return
    */
+  public DEdge findInternalLinkEdge(DDiagram pDiagram, EdgeTarget sourceNode, EdgeTarget targetNode, EdgeMapping mapping) {
```

Comment:
Since source/target does not make a lot of sense here, could we just name the variables "firstPort" and "secondPort" (just an idea)? Could you also add a comment for this method that it is supposed to always return the same internal link between 2 ports regardless of the creation order of interal links ?